### ISSUE TITLE
Add documentation schema

### ISSUE THREAD

**Issue by dingo-d** [squizlabs/PHP_CodeSniffer#3585]
In WordPress Coding Standards I've started to work on adding [xsd schema](https://github.com/WordPress/WordPress-Coding-Standards/pull/2037) for the phpcs documentation.

If you're up for it I can move the PR on this repo and we can then work on it here.

Let me know if you would like to have this in the current repo, or if we should just offload it to PHPCSUtils so that external standards ca use it if they want to.

Current xsd schema (passing the validation on WPCS) looks like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified">
    <xs:element name="documentation">
        <xs:complexType>
            <xs:sequence maxOccurs="unbounded">
                <xs:group ref="rulegroup"/>
            </xs:sequence>
            <xs:attribute name="title" use="required" type="titleType"/>
            <xs:anyAttribute processContents="lax"/>
        </xs:complexType>
    </xs:element>

    <xs:group name="rulegroup">
        <xs:sequence>
            <xs:element name="standard" type="standardType"/>
            <xs:element name="code_comparison" type="code_comparisonType" maxOccurs="unbounded"/>
            <xs:any minOccurs="0"/>
        </xs:sequence>
    </xs:group>

    <xs:simpleType name="titleType">
        <xs:restriction base="xs:string">
            <xs:maxLength value="58"/>
        </xs:restriction>
    </xs:simpleType>

    <xs:complexType name="code_comparisonType">
        <xs:sequence>
            <xs:element name="code" type="codeType" maxOccurs="2" minOccurs="2"/>
        </xs:sequence>
    </xs:complexType>

    <xs:complexType name="codeType">
        <xs:simpleContent>
            <xs:extension base="xs:string">
                <xs:attribute name="title" use="required" type="xs:string"/>
            </xs:extension>
        </xs:simpleContent>
    </xs:complexType>

    <xs:complexType name="standardType">
        <xs:simpleContent>
            <xs:extension base="xs:string">
                <xs:anyAttribute/>
            </xs:extension>
        </xs:simpleContent>
    </xs:complexType>
</xs:schema>
```

**Comment by gsherwood** [comment]
The documentation format isn't a high priority for me, so best to keep this external. I may consider removal of the whole doc schema in the future.

**Comment by jrfnl** [comment]
@gsherwood Thanks for letting us know. We'll get this sorted shortly and will publish the URL on which the XSD will be available here before closing the ticket (in case more people are interested in using it).

If you do decide to remove the docs generators in the future, would you be open to moving the code to a separate package which external standards could then optionally require ? If so, let's stay in touch about that. I'd be happy to host it in the [PHPCSStandards](http://github.com/phpcsstandards) organisation.

**Comment by jrfnl** [comment]
FYI: the XSD schema PR has been merged and will be included in the PHPCSDevTools 2.0.0 release (expected within the next few weeks).

**Comment by jrfnl** [comment]
Just to link the issues together - I just realized this is a near duplicate of #2489
